### PR TITLE
[RFC] Added I2C4 Support in IMX I2C Driver.

### PR DESCRIPTION
**Pull request by DevendraDevadiga**
Added the I2C4 support in IMX I2C Driver.
Signed-off-by: Devendra Devadiga <<devendradevadiga01@gmail.com>>

### CODE REVIEW

**Comment by DevendraDevadiga** [comment]
Dear Sir,
Any comment on this pull request?
Regards,
Devendra Devadiga

File: core/drivers/imx_i2c.c [hunk 1]
```diff
@@ -59,17 +62,24 @@ static struct io_pa_va i2c_bus[3] = {
...
+#if defined(I2C4_BASE)
+	[3] = { .pa = I2C4_BASE, },
+#endif
 #endif
 };
 
 static struct imx_i2c_clk {
 	struct io_pa_va base;
 	uint32_t i2c[ARRAY_SIZE(i2c_bus)];
+#if defined(CFG_MX6ULL)
```
Comment (jforissier) [inline core/drivers/imx_i2c.c:74]: This change doesn't seem related to the main purpose (add I2C4 support)? Therefore,  it should be moved to a separate commit.

**Review by jforissier — CHANGES_REQUESTED** [review]
Hi @DevendraDevadiga ,

Thank you for the patch!
- Please see my comment below
- For the commit subject the imperative mood is preferred: "Add I2C4 support in IMX I2C driver".
- Some text is needed in the commit description.
- Your Signed-off-by: is required (see [here](https://optee.readthedocs.io/en/latest/general/contribute.html)).

Tagging @OP-TEE/plat-imx for any platform-specific comments.

**Comment by DevendraDevadiga** [comment]
Dear Sir,
Thanks for the comment.
Updated the patch as per suggestion.

Regards,
Devendra Devadiga

**Comment by jforissier** [comment]
This looks OK to me in general but I have no idea about the platform specifics so any plat-imx reviewer please feel free to comment.

`Acked-by: Jerome Forissier <jerome@forissier.org>`

**Review by jforissier — APPROVED** [review]

**Comment by jforissier** [comment]
@DevendraDevadiga please squash the commits into a single one and add my Acked-by: tag to the commit description.

**Comment by clementfaure** [comment]
Hi @DevendraDevadiga ,
I would split your work into two commits,

One for the register and offset values ```imx: add I2C4 registers values for imx...``` on which you have my ```Reviewed-by: Clement Faure <clement.faure@nxp.com>```
The other one for the ```imx_i2c``` driver where you can ask @ldts for a review .

**Comment by ldts** [comment]
also it seems you found a bug on the imx6.h (I2C3 address).
would you mind having that as a separate patch as well?

**Comment by ldts** [comment]
please squash the commits in 3 separate ones so they can be reviewed (at the moment you are trying to merge 12 commits on your PR)
also please separate the commit header from the message (seems they are on the same line)

**Comment by ldts** [comment]
please squash the commits in 3 separate ones so they can be reviewed (at the moment you are trying to merge 12 commits on your PR)
also please separate the commit header from the message (seems they are on the same line)

**Comment by ldts** [comment]
check the section on squashing commits: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History
one you have the three commits that we need (with a title and some comment on the body), just force push it to the github branch so history is rewritten.

**Comment by jforissier** [comment]
@ldts +1

@DevendraDevadiga  a summary of the procedure is given at https://optee.readthedocs.io/en/latest/general/contribute.html#finalizing-your-contribution

**Comment by DevendraDevadiga** [comment]
I am facing issue in git squashing commands. So closing this request. Sorry for inconvenience caused.

**Comment by DevendraDevadiga** [comment]
I shared the required patch files.
[optee-imx-i2c4-patch.zip](https://github.com/OP-TEE/optee_os/files/6756595/optee-imx-i2c4-patch.zip)

**Comment by ldts** [comment]
@DevendraDevadiga let me try creating the branch for you, please do not give up. if you can pull an test that would be great. 
then just fork it and send the PR again? I can then take it from there if you want 
https://github.com/ldts/optee_os/commits/i2c4-devendra

**Pull request closed without merging**